### PR TITLE
refacto(tests): update use case tests

### PR DESCRIPTION
**Pull request by killianherbunot**

### CODE REVIEW

**Review by nnguon — APPROVED** [review]

**Review by thomasgataly — APPROVED** [review]

File: src/Skeleton/Tests/BusinessRules/UseCases/GenericUseCaseTest.php.twig [hunk 1]
```diff
@@ -10,27 +10,16 @@ use PHPUnit\Framework\TestCase;
 
 class {{ skeletonModel.shortName }} extends TestCase
```
Comment (arnaud-23) [inline src/Skeleton/Tests/BusinessRules/UseCases/GenericUseCaseTest.php.twig:11]: final

File: src/Skeleton/Tests/BusinessRules/UseCases/GenericUseCaseTest.php.twig [hunk 2]
```diff
@@ -10,27 +10,16 @@ use PHPUnit\Framework\TestCase;
...
-     * @var {{ skeletonModel.genericUseCaseShortName }}
-     */
-    private $useCase;
+    private {{ skeletonModel.genericUseCaseShortName }} $useCase;
 
-    /**
-     * @test
-     */
+    /** @test */
     final public function {{ skeletonModel.genericUseCaseTestMethod }}ReturnResponse(): void
```
Comment (arnaud-23) [inline src/Skeleton/Tests/BusinessRules/UseCases/GenericUseCaseTest.php.twig:18]: not final if class is final

File: tests/Fixtures/Classes/Tests/BusinessRules/UseCases/Domain/SubDomain/GenericUseCaseTest.php [hunk 3]
```diff
@@ -14,27 +14,16 @@
 
 class GenericUseCaseTest extends TestCase
```
Comment (arnaud-23) [inline tests/Fixtures/Classes/Tests/BusinessRules/UseCases/Domain/SubDomain/GenericUseCaseTest.php:15]: final

File: tests/Fixtures/Classes/Tests/BusinessRules/UseCases/Domain/SubDomain/GenericUseCaseTest.php [hunk 4]
```diff
@@ -14,27 +14,16 @@
...
-
-    /**
-     * @test
-     */
+    private GenericUseCaseRequestDTO $request;
+
+    private GenericUseCase $useCase;
+
+    /** @test */
     final public function genericUseCaseReturnResponse(): void
```
Comment (arnaud-23) [inline tests/Fixtures/Classes/Tests/BusinessRules/UseCases/Domain/SubDomain/GenericUseCaseTest.php:22]: not final

File: tests/Fixtures/Classes/Tests/BusinessRules/UseCases/Domain/SubDomain/GetFunctionalEntitiesTest.php [hunk 5]
```diff
@@ -24,19 +24,11 @@ class GetFunctionalEntitiesTest extends TestCase
...
-     * @var GetFunctionalEntities
-     */
-    private $useCase;
+    private GetFunctionalEntities $useCase;
 
-    /**
-     * @test
-     */
+    /** @test */
     final public function withoutFunctionalEntitiesReturnEmpty(): void
```
Comment (arnaud-23) [inline tests/Fixtures/Classes/Tests/BusinessRules/UseCases/Domain/SubDomain/GetFunctionalEntitiesTest.php:32]: not final but class should be final

File: tests/Fixtures/Classes/Tests/BusinessRules/UseCases/Domain/SubDomain/GetFunctionalEntityTest.php [hunk 6]
```diff
@@ -42,9 +36,7 @@ class GetFunctionalEntityTest extends TestCase
         $this->useCase->execute($this->request);
     }
 
-    /**
-     * @test
-     */
+    /** @test */
     final public function getFunctionalEntityShouldReturnResponse(): void
```
Comment (arnaud-23) [inline tests/Fixtures/Classes/Tests/BusinessRules/UseCases/Domain/SubDomain/GetFunctionalEntityTest.php:40]: not final but class should be final

**Review by arnaud-23 — CHANGES_REQUESTED** [review]

**Review by fredericjaume-oc — APPROVED** [review]

**Pull request merged**